### PR TITLE
redfish-core: Core: Enabled the isolation (aka guard) feature

### PR DESCRIPTION
**Pull request by deepakala-k**
- In an OpenPOWER based system, a user can isolate the Processor Core from the next boot of the system so add the support for same in this patch to do isolation through redfish interface.

- In Redfish, the "Enabled" property (which is added in Processor schema) to enable or disable the respective resource so using the same property for isolation feature.
  - The "Enabled" property was added in Processor.v1_12_0 version so bumped the processor schema version in all places.

- The "Enabled" Redfish property is mapped with "xyz.openbmc_project.Object.Enable::Enabled" dbus property.

- In this patch, added the utility function for hardware isolation so any redfish resource can use that utility function if that redfish resource will support hardware isolation for the system.

Note: The hardware isolation for the Processor Core is enabled on
      top of the below patch which will be used to model the
      Processor Core in redfish.
      https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/38570

Other Reference:

- Design document for hardware isolation (aka guard) https://github.com/openbmc/docs/blob/master/designs/guard-on-bmc.md

- DBus interface for "Enabled" https://github.com/openbmc/phosphor-dbus-interfaces/blob/master/yaml/ \ xyz/openbmc_project/Object/Enable.interface.yaml

- DBus interface for isolate hardware https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-dbus-interfaces/ \ +/43533

Tested:

- Disabled the particular Processor Core by using below command.
```
curl -k -H "X-Auth-Token: $bmc_token" -X PATCH \
https://${bmc}/redfish/v1/Systems/system/Processors/cpu0/SubProcessors/core0 \
-d '{"Enabled" : false }'
{
  "@Message.ExtendedInfo": [
    {
      "@odata.type": "#Message.v1_1_1.Message",
      "Message": "Successfully Completed Request",
      "MessageArgs": [],
      "MessageId": "Base.1.8.1.Success",
      "MessageSeverity": "OK",
      "Resolution": "None"
    }
  ]
}
```

- Disabled the isolated Processor Core by using below command.
```
curl -k -H "X-Auth-Token: $bmc_token" -X PATCH \
https://${bmc}/redfish/v1/Systems/system/Processors/cpu0/SubProcessors/core0 \
-d '{"Enabled" : false }'
{
  "Core@Message.ExtendedInfo": [
    {
      "@odata.type": "#Message.v1_1_1.Message",
      "Message": "The requested resource of type @odata.id with the property Core \
                  with the value core0 already exists.",
      "MessageArgs": [
        "@odata.id",
        "Core",
        "core0"
      ],
      "MessageId": "Base.1.8.1.ResourceAlreadyExists",
      "MessageSeverity": "Critical",
      "Resolution": "Do not repeat the create operation as the resource has already \
                    been created."
    }
  ]
}
```

- Enabled the particular isolated Processor Core by using below command.
```
curl -k -H "X-Auth-Token: $bmc_token" -X PATCH \
https://${bmc}/redfish/v1/Systems/system/Processors/cpu0/SubProcessors/core0 \
-d '{"Enabled" : true }'
{
  "@Message.ExtendedInfo": [
    {
      "@odata.type": "#Message.v1_1_1.Message",
      "Message": "Successfully Completed Request",
      "MessageArgs": [],
      "MessageId": "Base.1.8.1.Success",
      "MessageSeverity": "OK",
      "Resolution": "None"
    }
  ]
}
```

### CODE REVIEW

**Comment by deepakala-k** [comment]
This commit was missed out while rebasing which resulted in the below defect. https://jazz07.rchland.ibm.com:13443/jazz/web/projects/CSSD#action=com.ibm.team.workitem.viewWorkItem&id=638768

File: redfish-core/lib/processor.hpp [hunk 1]
```diff
@@ -1110,6 +1110,33 @@ inline void afterGetCpuCoreDataByService(
...
+                {
+                    if (propName == "Enabled")
+                    {
+                        const bool* enabled = std::get_if<bool>(&propValue);
+                        if (enabled == nullptr)
+                        {
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+                        asyncResp->res.jsonValue["Enabled"] = *enabled;
```
Comment (baemyung) [inline redfish-core/lib/processor.hpp:1138]: Can the loop do break here to skip the other properties?
```
                for (const auto& [propName, propValue] : properties)
                {
...
                        asyncResp->res.jsonValue["Enabled"] = *enabled;
                        break;     <--
                }
```

**Review by jeaaustx — COMMENTED** [review]
Looks good to me. I agree with Myung's comment on breaking out of the loop once the 'Enabled" property has been found.

File: redfish-core/lib/processor.hpp (hunk 1 above)
Comment (deepakala-k) [reply redfish-core/lib/processor.hpp:1138]: done

**Pull request merged**